File: SDKs/googleapis.com/secretmanager/v1/python/src/sdk/models/shared/secretpayload.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class SecretPayload:
    r"""SecretPayload
    A secret payload resource in the Secret Manager API. This contains the sensitive secret payload that is associated with a SecretVersion.
    """
    
    data: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    data_crc32c: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataCrc32c') }})
    

File: SDKs/googleapis.com/secretmanager/v1beta1/java/lib/src/main/java/openapisdk/models/shared/SecretPayload.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * SecretPayload
 * A secret payload resource in the Secret Manager API. This contains the sensitive secret data that is associated with a SecretVersion.
**/
public class SecretPayload {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("data")
    public String data;
    public SecretPayload withData(String data) {
        this.data = data;
        return this;
    }
}